couchbase/datadog_checks/couchbase/couchbase.py

```diff
@@ -402,3 +409,73 @@ def extract_seconds_value(self, value):
             unit = 'us'
 
         return float(val) / TO_SECONDS[unit]
+
+    def _collect_index_stats_metrics(self):
+        url = urljoin(self._index_stats_url, INDEX_STATS_METRICS_PATH)
+        try:
+            data = self._get_stats(url)
+        except requests.exceptions.RequestException as e:
+            msg = "Error accessing the Index Statistics endpoint: %s: %s" % (url, str(e))
+            self.log.warning(str(e))
+            self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.CRITICAL, self._tags, msg)
+            return
+
+        self.service_check(INDEX_STATS_SERVICE_CHECK_NAME, AgentCheck.OK, self._tags)
+
+        for keyspace in data:
+            if keyspace == "indexer":
+                # The indexer object provides metric about the index node
+                for mname, mval in data.get(keyspace).items():
+                    self._submit_index_node_metrics(mname, mval, self._tags)
+            else:
+                index_tags = self._extract_index_tags(keyspace) + self._tags
+                for mname, mval in data.get(keyspace).items():
+                    self._submit_per_index_metrics(mname, mval, index_tags)
+
+    def _extract_index_tags(self, keyspace):
+        # Index Keyspaces can come in different formats:
+        # partition, bucket:index_name, bucket:collection:index_name, bucket:scope:collection:index_name
+        # For variations missing the scope and collection, they refer to the default scope and collection respectively
+        # https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/createprimaryindex.html#keyspace-ref
+        tag_arr = keyspace.split(":")
+        if len(tag_arr) == 2:
+            bucket, index_name = tag_arr
+            scope, collection = ["default", "default"]
+        elif len(tag_arr) == 3:
+            bucket, collection, index_name = tag_arr
+            scope = 'default'
+        elif len(tag_arr) == 4:
+            bucket, scope, collection, index_name = tag_arr
+        else:
+            # Catch all incase the keyspace has either none or more than 3 separators(':')
+            # There is a documented example of partition-num being a possible keyspace:
+            # https://docs.couchbase.com/server/current/rest-api/rest-index-stats.html#_get_index_stats
+            # But we shouldn't encouter that with the current version of the check
```

Comment:
Which specific version is `current version` referencing?

Reply:
I agree sounds unclear here. What I was trying to refer to is the new version of the check after it's been merge. But I don't know how the naming convention for the check versions work. I can try rewording it though. Any ideas? Thanks in advance!

Reply:
Hm, I would word is as `version 1.19.0+ of the Couchbase check`.